extension.js

```diff
@@ -1,57 +1,160 @@
-const vscode = require('vscode');
+const { Location, Range, Uri, Position, workspace, commands, languages } = require('vscode');
 const path = require('path');
 
 class ReferenceProvider {
 
+    constructor() {
+        this.childWord = '';
+    }
+
+     /**
+     * Remove all white space and new lines from str
+     * @param {String} str
+     * @return {String}
+     */
+    stripWhiteSpaceAndNewLines(str) {
+        return (str + '').replace(/\s|\r|\n/g, '');
+    }
+
     /**
      * Get a require or define statement from string
      * @param {String} str
      * @return {String} 
      */
     getRequireOrDefineStatement(str) {
-        const match = /(define|require)\s*\(([^)]*)/gi.exec(str + "");
+        str = this.stripWhiteSpaceAndNewLines(str);
+        const match = /(define|require).*{/i.exec(str);
         return match && match[0] || null;
     }
 
     /**
+     * Get partial code from require/define statement until given line and character
+     * @param {VSCODE Document} document
+     * @param {Int} line to stop
+     * @param {Int} characterIndex to stop
+     * @return {String} substring for given document ending at given line and character and starting at last define or require
+     */
+    getRequireOrDefineCodeUntilCharacterIndex(document, line, characterIndex) {
```

Comment:
I get what your saying but would this not cause an outdated list real fast if someone is developing? Adding a newline would already require a complete rescan of the file.